test/fail_compilation/test20549.d

```diff
@@ -0,0 +1,5 @@
+module test;
```

Comment:
This should have a `TEST_OUTPUT` section checking for the correct error message.

Reply:
The error message is not the thing to be tested.

Reply:
All test that yield error messages should provide that section because its assumed to be silent otherwise. IIRC this is currently not enforced but will become the default once the remaining tests are updated.

Reply:
IIRC there were only a handful of tests left as they had a massive list of output, maybe they can be ignored manually?

Reply:
The point is not only to test the error message, but to make sure the assumptions made by the test hold.

Someone sometime later could change some code in DMD, and lead to the test case pass *for the wrong reason*
(here "passing" == "not compiling"). That could potentially mean the regression could sneak back because something unrelated in the test now fails.
And while it's unlikely with a 3 lines test, it's still a good habit to have. (On a side note, there also was some effort to have TEST_OUTPUT set as required IIRC).

Reply:
Sorry for not knowing all the test rules, there should be a test category for "not crash only".
Now thinking I should have written the code in a trait-compiles on the compilable category, much clever.

Reply:
`fail_compilation` is fine as it assigns a `module` to an `enum`.
Anyway, I assume the explanation concerning is in the tests README.md (or should be added there if it is missing)